source/MaterialXRuntime/RtCoreIo.cpp

```diff
@@ -521,6 +524,46 @@ void RtCoreIo::read(const DocumentPtr& doc, RtCoreIo::ReadFilter filter)
     }
 }
 
+void RtCoreIo::loadLibraries(const StringVec& libraryPaths, FileSearchPath& searchPaths, const RtToken& referenceName)
+{
+    // Always add in local search paths
+    FilePath currentPath(FilePath::getCurrentPath());
+    FilePath parentCurrentPath = currentPath.getParentPath();
+    const FilePath librariesPath("libraries");
+    FilePath fullPath(currentPath / librariesPath);
+    if (!fullPath.exists())
+    {
+        fullPath = parentCurrentPath / librariesPath;
+        if (fullPath.exists())
+        {
+            searchPaths.append(fullPath);
```

Comment:
Seems a little strange to me that the search paths get appended to. It means I pass in to the function a set of search paths, but when the function returns I have more search paths then I passed in to the searchPaths variable.

Reply:
Yeah, it is a bit weird. I should create a local copy.